core/src/test/java/org/apache/calcite/util/format/postgresql/PostgresqlDateTimeFormatterTest.java

```diff
@@ -1311,6 +1314,610 @@ void testEraLowerCaseWithDots(String pattern) {
+        PostgresqlDateTimeFormatter.toTimestamp("01", "YY", TIME_ZONE));
+    assertEquals(
+        createDateTime(2001, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("1", "YY", TIME_ZONE));
+    assertEquals(
+        createDateTime(2024, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("24", "YY", TIME_ZONE));
+  }
+
+  @Test void testToTimestampY() throws Exception {
+    assertEquals(
+        createDateTime(2001, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("1", "Y", TIME_ZONE));
+    assertEquals(
+        createDateTime(2001, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("1", "Y", TIME_ZONE));
+    assertEquals(
+        createDateTime(2004, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("4", "Y", TIME_ZONE));
+  }
+
+  @Test void testToTimestampIYYY() throws Exception {
+    assertEquals(
+        createDateTime(1, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("0001", "IYYY", TIME_ZONE));
+    assertEquals(
+        createDateTime(1, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("1", "IYYY", TIME_ZONE));
+    assertEquals(
+        createDateTime(2024, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("2024", "IYYY", TIME_ZONE));
+  }
+
+  @Test void testToTimestampIYY() throws Exception {
+    assertEquals(
+        createDateTime(2001, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("001", "IYY", TIME_ZONE));
+    assertEquals(
+        createDateTime(2001, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("1", "IYY", TIME_ZONE));
+    assertEquals(
+        createDateTime(1987, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("987", "IYY", TIME_ZONE));
+  }
+
+  @Test void testToTimestampIY() throws Exception {
+    assertEquals(
+        createDateTime(2001, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("01", "IY", TIME_ZONE));
+    assertEquals(
+        createDateTime(2001, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("1", "IY", TIME_ZONE));
+    assertEquals(
+        createDateTime(2024, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("24", "IY", TIME_ZONE));
+  }
+
+  @Test void testToTimestampI() throws Exception {
+    assertEquals(
+        createDateTime(2001, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("1", "I", TIME_ZONE));
+    assertEquals(
+        createDateTime(2001, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("1", "I", TIME_ZONE));
+    assertEquals(
+        createDateTime(2004, 1, 1, 0, 0, 0, 0),
+        PostgresqlDateTimeFormatter.toTimestamp("4", "I", TIME_ZONE));
+  }
+
+  @Test void testToTimestampBCAD() throws Exception {
+    assertEquals(
+        0,
+        PostgresqlDateTimeFormatter.toTimestamp("1920BC", "YYYYBC", TIME_ZONE)
```

Comment:
the rest of Calcite has trouble with BC dates, but that's no reason not to make this work